### PR TITLE
add vslib and yascreen to build system

### LINKED ISSUE #2: setting up an autoconf and automake file
https://github.com/cade-vs/vfu-dist/blob/036fe29881bde07605f104e16a3e1f97a005ac72/TODO#L8

I can set up an automake and autoconf file, which would provide a `./configure` script and Makefiles that are more portable than your current setup. I would need to go over a few details with you first before I start. Let me know if you're interested.

### PR DESCRIPTION
**Pull request by andy5995**
closes #2

### CODE REVIEW

File: build-aux/test-driver [hunk 1]
```diff
@@ -0,0 +1,148 @@
+#! /bin/sh
+# test-driver - basic testsuite driver script.
```
Comment (andy5995) [inline build-aux/test-driver:2]: This file was added automatically when I ran `autoreconf -ivf` after adding a Makefile to the 't' directory. It's used when `make check` is run.

File: configure.ac [hunk 2]
```diff
@@ -5,7 +5,7 @@ AC_INIT([vfu], [4.18], [https://github.com/cade-vs/vfu/issues],,[http://cade.dat
 AC_CONFIG_AUX_DIR([build-aux])
 AM_INIT_AUTOMAKE([subdir-objects foreign no-dependencies])
 AC_PREREQ(2.69)
-AC_CONFIG_HEADERS([vfu/config.h:config.in])
+AC_CONFIG_HEADERS([config.h:config.in])
```
Comment (andy5995) [inline configure.ac:8]: vslib uses this so I moved it to a common directory

File: yas-libs/Makefile.am [hunk 3]
```diff
@@ -0,0 +1,3 @@
+noinst_LIBRARIES = libyascreen.a
```
Comment (andy5995) [inline yas-libs/Makefile.am]: Using this file in this directory means that a Makefile.am doesn't need to be added to the yascreen directory or repo.

**Comment by cade-vs** [comment]
ok I merged everything, thanks! now I'll need some time to check it :)

**Comment by andy5995** [comment]
Hi @cade-vs .

This one still needs merging https://github.com/cade-vs/vslib/pull/5 or I can do something similar to what I did for yas-libs and put the vslib Makefile into vfu-dist

**Pull request merged**